### DISCUSSION TITLE
rheoFoam

### DISCUSSION THREAD

**Discussion by am2021-create** [fppimenta/rheoTool#46] [Q&A]
Hi fppimenta ,

May you kindly assist me, I am using rheoTool. Trying to run the rheoFoam solver using the one of the tutorials provided GiesekusLog. In the solver provided they are simulating the fluid flow in a channel. I want to simulate this using a wedge, I tried using one of the blockMeshDict files from the rheoFoam GNF - Newtonian tutorial .It gives me the below errors all and I did was to change the geomerty from a channel to a wedge. I further tried to create my own circular pipe blockMesh but it still gives me the below error

// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
Create time

Create mesh for time = 0

Reading field p

Reading field U

Reading/calculating face flux field phi

No MRF models present

No finite volume options present
Using openFoamSolver solver for p
Using openFoamSolver solver for U
Selecting constitutive equation GiesekusLog
Selecting thermoFunction: Constant
Selecting thermoFunction: Constant

SIMPLE: No convergence criteria found

SIMPLE: No convergence criteria found

SIMPLE: Operating solver in transient mode with 1 outer corrector
SIMPLE: Operating solver in PISO mode

Courant Number mean: 0.166666666667 max: 20.0000000003

Starting time loop

Courant Number mean: 0.166666666667 max: 20.0000000003
Time = 0.05

Inner iteration: 0

DICPCG: Solving for Ux, Initial residual = 1, Final residual = 8.85012301813e-11, No Iterations 9
DICPCG: Solving for Uy, Initial residual = 1.68230678437e-35, Final residual = 1.68230678437e-35, No Iterations 0
DICPCG: Solving for Uz, Initial residual = 2.61013189937e-34, Final residual = 2.61013189937e-34, No Iterations 0
DICPCG: Solving for p, Initial residual = 1, Final residual = 1.69922271366e-11, No Iterations 128
time step continuity errors : sum local = 1.53609117535e-09, global = 1.54235196027e-11, cumulative = 1.54235196027e-11
DILUPBiCG: Solving for thetaxx, Initial residual = 1, Final residual = 4.73377676548e-14, No Iterations 5
DILUPBiCG: Solving for thetaxy, Initial residual = 0, Final residual = 0, No Iterations 0
DILUPBiCG: Solving for thetaxz, Initial residual = 0, Final residual = 0, No Iterations 0
DILUPBiCG: Solving for thetayy, Initial residual = 1, Final residual = 4.91394049142e-14, No Iterations 5
DILUPBiCG: Solving for thetayz, Initial residual = 0, Final residual = 0, No Iterations 0
DILUPBiCG: Solving for thetazz, Initial residual = 1, Final residual = 1.9270971297e-14, No Iterations 5
ExecutionTime = 0.08 s ClockTime = 0 s

Courant Number mean: 41.5991225201 max: 287.839559909
Time = 0.1

Inner iteration: 0

DICPCG: Solving for Ux, Initial residual = 0.34212118528, Final residual = 3.77858587337e-11, No Iterations 7
DICPCG: Solving for Uy, Initial residual = 0.184547365086, Final residual = 1.44042045095e-11, No Iterations 8
DICPCG: Solving for Uz, Initial residual = 0.737359389751, Final residual = 2.44027650758e-11, No Iterations 7
DICPCG: Solving for p, Initial residual = 0.00408777212477, Final residual = 8.09619199808e-12, No Iterations 126
time step continuity errors : sum local = 3.02672895644e-09, global = 2.47038699989e-11, cumulative = 4.01273896016e-11
DILUPBiCG: Solving for thetaxx, Initial residual = 0.260162746334, Final residual = 4.58120438555e-11, No Iterations 4
DILUPBiCG: Solving for thetaxy, Initial residual = 1, Final residual = 6.3196806269e-15, No Iterations 5
DILUPBiCG: Solving for thetaxz, Initial residual = 0.999904252247, Final residual = 3.19930763771e-11, No Iterations 3
DILUPBiCG: Solving for thetayy, Initial residual = 0.264513260497, Final residual = 1.77389571288e-14, No Iterations 5
DILUPBiCG: Solving for thetayz, Initial residual = 0.997980409006, Final residual = 7.08204831597e-11, No Iterations 4
DILUPBiCG: Solving for thetazz, Initial residual = 0.301114558716, Final residual = 8.78553338396e-12, No Iterations 4
#0 Foam::error::printStack(Foam::Ostream&) at ??:?
#1 Foam::sigFpe::sigHandler(int) at ??:?
#2 ? in "/lib/x86_64-linux-gnu/libpthread.so.0"
#3 ? in "/lib/x86_64-linux-gnu/libm.so.6"
#4 expf64 in "/lib/x86_64-linux-gnu/libm.so.6"
#5 Foam::constitutiveEq::calcEig(Foam::GeometricField<Foam::SymmTensor, Foam::fvPatchField, Foam::volMesh> const&, Foam::GeometricField<Foam::Tensor, Foam::fvPatchField, Foam::volMesh>&, Foam::GeometricField<Foam::Tensor, Foam::fvPatchField, Foam::volMesh>&) at ??:?
#6 Foam::constitutiveEqs::GiesekusLog::correct() at ??:?
#7 ? in "/home/machine/OpenFOAM/machine-7/platforms/linux64GccDPInt32Opt/bin/rheoFoam"
#8 __libc_start_main in "/lib/x86_64-linux-gnu/libc.so.6"
#9 ? in "/home/machine/OpenFOAM/machine-7/platforms/linux64GccDPInt32Opt/bin/rheoFoam"
Floating point exception (core dumped)

**Comment by fppimenta** [comment]
It's not a code issue.

Check your initial Co, it is ~20. This is because the pipe geom for the GNF tutorials has much smaller dimensions than the straight channel in GiesekusLog tutorial. De is therefore much higher and what you see is just numerical instability for that conditions.

Either redimension the pipe or adjust the time-step or physical properties.

Take care with blind copy-paste.

**Comment by am2021-create** [comment]
Dear Francisco Pimenta,

Thank you for your response

Oh okay, I see thank much appreciated. I will implement your advise

Regards
